src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -695,6 +715,28 @@ protected virtual void Dispose(bool finalize)
 
         // We're disposing this workspace.  Stop any work to update SG docs in the background.
         _updateSourceGeneratorsQueueTokenSource.Cancel();
+        _workQueueTokenSource.Cancel();
+    }
+
+    private async ValueTask ProcessWorkQueueAsync(ImmutableSegmentedList<Action> list, CancellationToken cancellationToken)
+    {
+        // Hop over to the right scheduler to execute all this work.
+        await Task.Factory.StartNew(() =>
+        {
+            foreach (var item in list)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This cancellation _could_ change behavior here -- before Workspace events would be queued and would continue, at least until the termination of the process. If that cancellation triggers at some earlier point now it might mean people won't get events. Practically, this would only matter during shutdown, and since we were queuing to the UI thread there was already a pretty good chance those might never fire -- there's really no difference between "we cancelled them and they never ran during shutdown" versus "the UI thread was never available during shutdown to run them anyways".

This is probably a small win for shutdown perf; but if we see any strange bugs appear during shutdown this might be a place to investigate. Probably good to see if it's fine, at least!

Reply:
yup.  i'm very ok with this behavior.  you dispose of the workspace, and outstanding work may not happen.